polars/polars-arrow/src/kernels/rolling/no_nulls.rs

```diff
@@ -95,13 +183,284 @@ where
+                window_size,
+                min_periods,
+                det_offsets_center,
+                compute_weighted_quantile,
+                weights,
+            )
+        }
+        (false, Some(weights)) => {
+            let values = as_floats(values);
+            rolling_apply_weighted_quantile(
+                values,
+                0.5,
+                QuantileInterpolOptions::Linear,
+                window_size,
+                min_periods,
+                det_offsets,
+                compute_weighted_quantile,
+                weights,
+            )
+        }
+    }
+}
+
+pub fn rolling_quantile<T>(
+    values: &[T],
+    quantile: f64,
+    interpolation: QuantileInterpolOptions,
+    window_size: usize,
+    min_periods: usize,
+    center: bool,
+    weights: Option<&[f64]>,
+) -> ArrayRef
+where
+    T: NativeType + std::iter::Sum<T> + std::cmp::PartialOrd + num::ToPrimitive,
+{
+    match (center, weights) {
+        (true, None) => rolling_apply_quantile(
+            values,
+            quantile,
+            interpolation,
+            window_size,
+            min_periods,
+            det_offsets_center,
+            compute_quantile,
+        ),
+        (false, None) => rolling_apply_quantile(
+            values,
+            quantile,
+            interpolation,
+            window_size,
+            min_periods,
+            det_offsets,
+            compute_quantile,
+        ),
+        (true, Some(weights)) => {
+            let values = as_floats(values);
+            rolling_apply_weighted_quantile(
+                values,
+                quantile,
+                interpolation,
+                window_size,
+                min_periods,
+                det_offsets_center,
+                compute_weighted_quantile,
+                weights,
+            )
+        }
+        (false, Some(weights)) => {
+            let values = as_floats(values);
+            rolling_apply_weighted_quantile(
+                values,
+                quantile,
+                interpolation,
+                window_size,
+                min_periods,
+                det_offsets,
+                compute_weighted_quantile,
+                weights,
+            )
+        }
+    }
+}
+
 pub(crate) fn compute_sum<T>(values: &[T]) -> T
 where
     T: std::iter::Sum<T> + Copy,
 {
     values.iter().copied().sum()
 }
 
+pub(crate) fn compute_quantile<T>(
+    values: &[T],
+    quantile: f64,
+    interpolation: QuantileInterpolOptions,
+) -> f64
+where
+    T: std::iter::Sum<T> + Copy + std::cmp::PartialOrd + num::ToPrimitive,
+{
+    if !(0.0..=1.0).contains(&quantile) {
+        panic!("quantile should be between 0.0 and 1.0");
+    }
+
+    let mut vals: Vec<f64> = values
+        .iter()
+        .copied()
+        .map(|x| NumCast::from(x).unwrap())
+        .collect();
+    vals.sort_by(|a, b| a.partial_cmp(b).unwrap());
+
+    let length = vals.len();
+
+    let mut idx = match interpolation {
```

Comment:
Maybe we can do the `idx` and the `f64` computation in a single pattern match? I think that will be a bit more performant.

Reply:
I'll make this change in my next PR if that's okay

Reply:
Sure!